src/xtopcom/xblockmaker/src/xunit_builder.cpp

```diff
@@ -49,31 +52,95 @@ xblock_ptr_t        xlightunit_builder_t::build_block(const xblock_ptr_t & prev_
-        build_para->set_error_code(xblockmaker_error_tx_execute);
-        return nullptr;
+    if (new_vm) {
+        for (auto const & tx : input_txs) {
+            xdbg("------>new vm, %s, %s, %d\n", tx->get_source_addr().c_str(), tx->get_target_addr().c_str(), tx->get_tx_subtype());
+            assert(!tx->get_transaction()->get_source_action().get_action_param().empty());
+            xdbg("------>new vm, %s, %zu, %s\n", tx->get_transaction()->get_source_action().get_action_name().c_str(), tx->get_transaction()->get_source_action().get_action_param().size(),
+                                                 tx->get_transaction()->get_source_action().get_action_param().c_str());
+        }
+
+        xassert(!cs_para.get_table_account().empty());
+        xassert(!cs_para.get_random_seed().empty());
+        xassert(cs_para.get_table_proposal_height() > 0);
+
+        auto * system_contract_manager = contract_runtime::system::xsystem_contract_manager_t::instance();
+        xassert(system_contract_manager != nullptr);
+        contract_vm::xaccount_vm_t vm(make_observer(system_contract_manager));
+
+        auto tx_address = common::xaccount_address_t{prev_block->get_account()};
+        auto _temp_header = base::xvblockbuild_t::build_proposal_header(prev_block.get());
+        auto proposal_bstate = make_object_ptr<base::xvbstate_t>(*_temp_header.get(), *prev_bstate.get());
+        std::map<common::xaccount_address_t, observer_ptr<base::xvbstate_t>> state_pack;
+        state_pack.insert(std::make_pair(tx_address, make_observer(proposal_bstate.get())));
+        auto result = vm.execute(input_txs, state_pack, cs_para);
+
+        lightunit_build_para->set_fail_txs(result.failed_tx_assemble);
+        lightunit_build_para->set_pack_txs(result.success_tx_assemble);
+        if (result.success_tx_assemble.size() == 0) {
+            build_para->set_error_code(xblockmaker_error_tx_execute);
+            return nullptr;
+        }
+
+        xlightunit_block_para_t lightunit_para;
+        lightunit_para.set_input_txs(result.success_tx_assemble);
+        assert(result.bincode_pack.count(tx_address));
+        assert(result.binlog_pack.count(tx_address));
+        lightunit_para.set_fullstate_bin(result.bincode_pack.at(tx_address));
+        lightunit_para.set_binlog(result.binlog_pack.at(tx_address));
+
+        base::xreceiptid_state_ptr_t receiptid_state = lightunit_build_para->get_receiptid_state();
+        alloc_tx_receiptid(result.success_tx_assemble, receiptid_state);
+        base::xvblock_t* _proposal_block = data::xblocktool_t::create_next_lightunit(lightunit_para, prev_block.get(), cs_para);
+        xblock_ptr_t proposal_unit;
+        proposal_unit.attach((data::xblock_t *)_proposal_block);
+        return proposal_unit;
+    } else {
+        for (auto const & tx : input_txs) {
+            xdbg("------>old vm, %s, %s, %d\n", tx->get_source_addr().c_str(), tx->get_target_addr().c_str(), tx->get_tx_subtype());
```

Comment:
could add `assert(tx->get_tx_type() != enum_xtransaction_type::xtransaction_type_run_contract);` here.
To prevent cons pack different type's tx in the furture, which would run inside old txexecutor logical.